### PR TITLE
Bed war DLC: Lance, shields and pillow carnage

### PR DESCRIPTION
**Pull request by TheBoondock**
## About The Pull Request
The pillow expansion that I've been promising this summer is here.

This pr brings about the pillow lance, the pillow shield, adds a fortify ability to the pillow suit and a feather effect when hitting with a pillow 

### Pillow lance
image

Boasting the best offensive ability of the three addition, the pillow lance deals 15 stamina damage but it is bulky and requires 2 hands to use. Its special ability allows the user to charge forward, with a range of 7 tiles. Each tiles charged will increase the stamina damage dealt to the victim each tiles adds 10dmg up to 70 stam dmg. Careful tho as it uses 60 stamina and immobilize you for a second after charging.

### Pillow shield
image

Your very own captain america shield! Pillow shield has a boomerang funcition, allowing you to hit your foe from afar, using up 20 stamina in the process (5 throws total) Pillow shield gain increase blocking chance against pillow weaponry and more so if it is wielded. Oh and you can bash your enemy with it too!

### Juggernaut Mode (pillow suit)
image

The pillownaut suit has become even more formidable. Its special ability allows the user to hunker down, slowing dramatically and immune to shove knockdown. The bump attack also hits harder than regular. Use this to break the defensive line.
## Why It's Good For The Game
More content for pillow warfare, with stamina based resource for combat making for more strategic decisions.

## TO-DO
- [ ] Map and add a crafting recipe for these items
- [x] Fixes known bugs (pillow shield hitting user on return, pillow hit effects not applying)
- [ ] Add some context messages for abilities and actions
## Changelog
:cl:
add: Added Pillow Lance, a 2 handed lance dealing 15 stamina damage per hit and is able to charge 7 tiles, consuming 60 stamina and dealing more damage the futher you travel.
add: Added Pillow Shield, a boomerang shield that you can throw to do 10 stamina damage, the shield gains bonus blocking chance against pillow weaponry and when wielded with both hands.
add: Added juggernaut mode to pillow suit, juggernaut mode slows down the user alot and makes them immune to shove knockdown, it also increase the bumpattack force.
add: Added a feather hit effect for pillow weaponry
/:cl:

### CODE REVIEW

**Comment by ChungusGamer666** [comment]
this BETTER have some special interactions with #76341 if that gets merged

**Comment by atlasle** [comment]
could you add a crate that contains like 10 pillows too, cause we're seriously going to be running out of pillows

**Comment by Elli-Skala** [comment]
What about the pillow lance has the duel weild as a option then a requirement so that we can weild a pillow shield and lance for a more balanced build

**Comment by Pickle-Coding** [comment]
An 80% block chance against stamina weapons for a wielded pillow shield seems insane. I'm not sure about having pillows be actually effective for actual (non-pillow) combat.

The block chance should strictly only apply to pillows and toys, otherwise every antagonist is going to bring a pillow.

**Comment by AtomTheProphet** [comment]
> An 80% block chance against stamina weapons for a wielded pillow shield seems insane. I'm not sure about having pillows be actually effective for actual (non-pillow) combat.
> 
> The block chance should strictly only apply to pillows and toys, otherwise every antagonist is going to bring a pillow.

Maybe it IS time for a new age of pillow-based warfare.

**Comment by TheBoondock** [comment]
> An 80% block chance against stamina weapons for a wielded pillow shield seems insane. I'm not sure about having pillows be actually effective for actual (non-pillow) combat.
> 
> The block chance should strictly only apply to pillows and toys, otherwise every antagonist is going to bring a pillow.

No no, it only gain up ward of 70% blocking chance if it's being hit with a pillow based weapon AND it is being wield 2 handed

**Comment by Pickle-Coding** [comment]
Oh my bad. Somehow I didn't see that check.

File: code/game/objects/items/shields.dm [hunk 1]
```diff
@@ -330,4 +330,51 @@
...
+		new /obj/effect/temp_visual/pillow_hit(hit_atom)
+		return
+
+
+/obj/item/shield/pillow/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text, final_block_chance, damage, attack_type, damage_type)
+	///Pillow shield gain bonus blocking for pillow weaponry and bonus blocking for wielding
+	if(damage_type == STAMINA)
+		if((hitby in countered_object) && HAS_TRAIT(src, TRAIT_WIELDED))
+			final_block_chance += 40
+			final_block_chance += 30
```
Comment (Pickle-Coding) [inline code/game/objects/items/shields.dm:373]: Why do it like this?

File: code/game/objects/items/pillow.dm [hunk 2]
```diff
@@ -141,6 +146,35 @@
...
+		end_fortify(user)
+
+
+/obj/item/clothing/suit/pillow_suit/proc/fortify(mob/living/user)
+	hunkered = TRUE
+	clothing_flags = BLOCKS_SHOVE_KNOCKDOWN
+	unstoppably_plushed.force += 10
+	user.add_movespeed_modifier(/datum/movespeed_modifier/pillow_fortify)
+	user.visible_message(span_alert("[user.name] hunkers down into a defensive stance!"))
+	START_PROCESSING(SSobj, src)
```
Comment (MrMelbert) [inline code/game/objects/items/pillow.dm:161]: Why process?

File: code/game/objects/items/pillow.dm [hunk 3]
```diff
@@ -141,6 +146,35 @@
...
+	if(hunkered)
+		end_fortify(user)
+
+
+/obj/item/clothing/suit/pillow_suit/proc/fortify(mob/living/user)
+	hunkered = TRUE
+	clothing_flags = BLOCKS_SHOVE_KNOCKDOWN
+	unstoppably_plushed.force += 10
+	user.add_movespeed_modifier(/datum/movespeed_modifier/pillow_fortify)
+	user.visible_message(span_alert("[user.name] hunkers down into a defensive stance!"))
```
Comment (MrMelbert) [inline code/game/objects/items/pillow.dm:160]:
```suggestion
	user.visible_message(span_alert("[user] hunkers down into a defensive stance!"))
```

File: code/game/objects/items/pillow.dm [hunk 4]
```diff
@@ -141,6 +146,35 @@
...
+/obj/item/clothing/suit/pillow_suit/proc/end_fortify(mob/living/user)
+	hunkered = FALSE
+	clothing_flags = null
+	unstoppably_plushed.force -= 10
+	user.remove_movespeed_modifier(/datum/movespeed_modifier/pillow_fortify)
+	STOP_PROCESSING(SSobj, src)
+	var/filter = user.get_filter(FORTIFY_FILTER)
+	animate(filter)
+	user.remove_filter(FORTIFY_FILTER)
+	user.visible_message(span_alert("[user.name] loosen up and goes into relax stance!"))
```
Comment (MrMelbert) [inline code/game/objects/items/pillow.dm:176]:
```suggestion
	user.visible_message(span_alert("[user] loosens up and relaxes a bit."))
```

File: code/game/objects/items/pillow.dm [hunk 5]
```diff
@@ -130,7 +132,10 @@
 	worn_icon = 'icons/mob/clothing/suits/pillow.dmi'
 	icon_state = "pillow_suit"
 	armor_type = /datum/armor/suit_pillow_suit
+	actions_types = list(/datum/action/item_action/pillow_fortify)
 	var/obj/item/pillow/unstoppably_plushed
+	var/hunkered = FALSE
+	var/outline_colour = "#eb0c07"
```
Comment (MrMelbert) [inline code/game/objects/items/pillow.dm:138]: Docs

File: code/game/objects/items/shields.dm [hunk 6]
```diff
@@ -330,4 +330,51 @@
...
+	icon_state = "pillow_shield"
+	inhand_icon_state = "pillow_shield"
+	block_chance = 10
+	max_integrity = 10
+	throw_range = 7
+	throw_speed = 1
+	force = 10
+	damtype = STAMINA
+	w_class = WEIGHT_CLASS_NORMAL
+	var/list/countered_object = list(/obj/item/spear/pillow, /obj/item/shield/pillow, /obj/item/pillow)
```
Comment (MrMelbert) [inline code/game/objects/items/shields.dm:345]: Docs, also static list?

File: code/game/objects/items/shields.dm [hunk 7]
```diff
@@ -330,4 +330,51 @@
...
+	)
+
+/obj/item/shield/pillow/attack(mob/living/target_mob, mob/living/user, params)
+	. = ..()
+	new /obj/effect/temp_visual/pillow_hit(target_mob)
+	playsound(target_mob, 'sound/items/pillow_hit2.ogg', 80)
+
+/obj/item/shield/pillow/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	. = ..()
+	var/caught = hit_atom.hitby(src, skipcatch = FALSE, hitpush = FALSE, throwingdatum = throwingdatum)
```
Comment (MrMelbert) [inline code/game/objects/items/shields.dm:362]: `.` is `TRUE` if it was caught, fairly certain you should not be calling `hitby` again.

File: code/game/objects/items/shields.dm [hunk 8]
```diff
@@ -330,4 +330,51 @@
...
+/obj/item/shield/pillow/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text, final_block_chance, damage, attack_type, damage_type)
+	///Pillow shield gain bonus blocking for pillow weaponry and bonus blocking for wielding
+	if(damage_type == STAMINA)
+		if((hitby in countered_object) && HAS_TRAIT(src, TRAIT_WIELDED))
+			final_block_chance += 40
+			final_block_chance += 30
+		else
+			final_block_chance = 10
+		owner.apply_damage(20, STAMINA)//block at your own risk
+	. = ..()
```
Comment (MrMelbert) [inline code/game/objects/items/shields.dm:377]:
```suggestion
	return ..()
```

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by nikothedude** [comment]
im pretty indifferent and even confused about the shield/spear (why does the shield have the boomerang trait) but oh my god i am rolling at the idea of seeing a goddamn pillow juggernaut shrugging off maintfu like its nothing

**Comment by ophaq** [comment]
Please add this so I can terrorize medbay when I'm bored not doing surgery.

**Comment by Loafin34** [comment]
> Oh my bad. Somehow I didn't see that check. It still gives a 10% block chance against batons though.

let the pillow man cook

**Comment by TheBoondock** [comment]
undrafting for review

**Comment by san7890** [comment]
mark ready for review when ci is passing

**Comment by Dawnseer** [comment]
Bring it back, it needs to happen!

**Comment by Nerev4r** [comment]
retvrn....

**Comment by Dawnseer** [comment]
Why are we still here... Just to suffer? Every night I can feel my sheets and my pillow, and even my pajamas... The bed I've lost... The plushies I've lost... It won't stop hurting. Like they're all still there. I'm the one who got caught up with Centcomm. A group above stations, even earthgov, and I was just a parasite below, sleeping on John Nanotrasen's bill.

**Comment by TheBoondock** [comment]
> Why are we still here... Just to suffer? Every night I can feel my sheets and my pillow, and even my pajamas... The bed I've lost... The plushies I've lost... It won't stop hurting. Like they're all still there. I'm the one who got caught up with Centcomm. A group above stations, even earthgov, and I was just a parasite below, sleeping on John Nanotrasen's bill.

I know I know, it will be in the game. The everything is essentially final and I just need the motivation to do pr it again.

**Pull request closed without merging**